tests/unit_tests/test_game_rater.py

```diff
@@ -9,211 +11,228 @@ class MockPlayer:
+
+    return [
+        {player1_id: RatingData(player1_outcome, player1_rating)},
+        {player2_id: RatingData(player2_outcome, player2_rating)},
+    ]
+
+
+@pytest.fixture
+def rating_data_2v2():
+    game_id = 111
+
+    player1_id = 1
+    player1_rating = Rating(1500, 500)
+    player1_outcome = GameOutcome.VICTORY
+
+    player2_id = 2
+    player2_rating = Rating(1400, 400)
+    player2_outcome = GameOutcome.DEFEAT
+
+    player3_id = 3
+    player3_rating = Rating(1300, 300)
+    player3_outcome = GameOutcome.DEFEAT
+
+    player4_id = 4
+    player4_rating = Rating(1200, 200)
+    player4_outcome = GameOutcome.DEFEAT
+
+    return [
+        {
+            player1_id: RatingData(player1_outcome, player1_rating),
+            player2_id: RatingData(player2_outcome, player2_rating),
+        },
+        {
+            player3_id: RatingData(player3_outcome, player3_rating),
+            player4_id: RatingData(player4_outcome, player4_rating),
+        },
+    ]
+
+
+def test_only_rate_with_two_parties():
+    one_party = [{1: GameOutcome.VICTORY}]
+    two_parties = [{1: GameOutcome.VICTORY}, {2: GameOutcome.DEFEAT}]
+    three_parties = [
+        {1: GameOutcome.VICTORY},
+        {2: GameOutcome.DEFEAT},
+        {3: GameOutcome.DEFEAT},
+    ]
 
-    rater = GameRater(players_by_team, outcome_py_player)
-    rating_groups = rater._get_rating_groups()
+    with pytest.raises(GameRatingError):
+        GameRater._check_rating_groups(one_party)
+
+    with pytest.raises(GameRatingError):
+        GameRater._check_rating_groups(three_parties)
 
-    assert len(rating_groups) == 2
-    assert {p1: Rating(1500, 500)} in rating_groups
+    GameRater._check_rating_groups(two_parties)
 
 
-def test_team_outcome():
-    p1, p2 = MockPlayer(), MockPlayer()
-    players_by_team = {2: [p1], 3: [p2]}
-    outcome_py_player = {p1: GameOutcome.VICTORY, p2: GameOutcome.DEFEAT}
+def test_team_outcome(rating_data_2v2):
+    victory_team = rating_data_2v2[0]
+    defeat_team = rating_data_2v2[1]
 
-    rater = GameRater(players_by_team, outcome_py_player)
-    assert rater._get_team_outcome([p1]) is GameOutcome.VICTORY
-    assert rater._get_team_outcome([p2]) is GameOutcome.DEFEAT
+    assert GameRater._get_team_outcome(victory_team) is GameOutcome.VICTORY
+    assert GameRater._get_team_outcome(defeat_team) is GameOutcome.DEFEAT
 
 
 def test_team_outcome_ignores_unknown():
-    p1, p2, p3, p4 = MockPlayer(), MockPlayer(), MockPlayer(), MockPlayer()
-    players_by_team = {2: [p1, p2], 3: [p3, p4]}
-    outcome_py_player = {
-        p1: GameOutcome.VICTORY,
-        p2: GameOutcome.UNKNOWN,
-        p3: GameOutcome.DEFEAT,
-        p4: GameOutcome.UNKNOWN
+    rating = Rating(1500, 500)
+    victory_team = {
+        1: RatingData(GameOutcome.VICTORY, rating),
+        2: RatingData(GameOutcome.UNKNOWN, rating),
+    }
+    defeat_team = {
+        1: RatingData(GameOutcome.DEFEAT, rating),
+        2: RatingData(GameOutcome.UNKNOWN, rating),
```

Comment:
Doesn't matter for the test, but in practice these player id's would be different from the ones on `victory_team` right?

Reply:
Right!